compiler/testData/diagnostics/tests/regressions/kt7804.fir.kt

```diff
@@ -18,7 +18,7 @@ fun <T> test(v: T): T {
 
 fun <T> test2(v: T): T {
     val a = if (v !is A) {
-        foo(v) as T
+        foo(v) <!USELESS_CAST!>as T<!>
     }
     else {
         v as T
```

Comment:
FE 1.0 warns `USELESS_CAST` here too, but reaching here means `v is A`, and thus `as T` seems still meaningful, but maybe I'm wrong.